crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -155,6 +157,79 @@ impl<'a> Checker<'a> {
             last_stmt_end: TextSize::default(),
         }
     }
+
+    // Determines whether a member access expression is referring to a
+    // member of a class (either a class or instance member). This will
+    // typically take the form "self.x" or "cls.x".
+    fn get_member_access_info(&self, value: &Expr) -> Option<MemberAccessInfo> {
```

Comment:
This is inspired by https://github.com/microsoft/pyright/blob/923bc3788cd35d90db828a1114dda127e3f1cced/packages/pyright-internal/src/analyzer/binder.ts#L3961